lib/simulator-xcode-8.js

```diff
@@ -53,6 +53,29 @@ class SimulatorXcode8 extends SimulatorXcode7 {
     }
   }
 
+  async openUrl (url) {
+    const SAFARI_STARTUP_TIMEOUT = 15 * 1000;
+
+    if (!await this.isRunning()) {
+      throw new Error(`Tried to open ${url}, but Simulator is not in Booted state`);
+    }
+    const launchTimestamp = process.hrtime();
+    try {
+      await waitForCondition(async () => {
+        try {
+          await exec('bash', ['-c', `xcrun simctl launch ${this.udid} com.apple.mobilesafari ${url}`]);
+          return true;
+        } catch (e) {
+          log.error(e);
+        }
+        return false;
+      }, {waitMs: SAFARI_STARTUP_TIMEOUT, intervalMs: 500});
+    } catch (err) {
+      log.errorAndThrow(`Safari cannot be started after ${process.hrtime(launchTimestamp)[0]} seconds`);
```

Comment:
Yes. Some indication of what the error is. Most of the time it will be that the browser didn't open and timed out. But there have been cases where this sort of pattern has hidden some configurational error or another, and it would _never_ work but was not apparent.